hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/GroupsLegendLayout.java

```diff
@@ -185,11 +211,11 @@ public void populateGroupsLegendByValidation(final RolloutGroupsValidation valid
         if (validation == null) {
             return;
         }
-        List<Long> targetsPerGroup = validation.getTargetsPerGroup();
+        final List<Long> targetsPerGroup = validation.getTargetsPerGroup();
         final long unassigned = validation.getTotalTargets() - validation.getTargetsInGroups();
-        final int labelsToUpdate = (unassigned > 0) ? (groupsLegend.getComponentCount() - 1)
+        final int labelsToUpdate = (unassigned > 0) ? (getGroupsWithoutToBeContinuedLabel(groups.size()) - 1)
```

Comment:
you already subtract one (-1) in getGroupsWithoutToBeContinuedLabel() method, check if it is necessary to do additional subtraction

Reply:
Yes it is